types/time_test.go

```diff
@@ -84,7 +86,10 @@ func TestDateTimeUnmarshalJSON(t *testing.T) {
 		jsonError  *json.UnmarshalTypeError
 		parseError *time.ParseError
 	)
-	require.ErrorAs(t, json.Unmarshal([]byte(`69`), &testTime), &jsonError)
+	err := json.Unmarshal([]byte(`69`), &testTime)
+	require.Truef(t, errors.As(err, &jsonError) || strings.Contains(err.Error(), "Mismatch type string with value number"),
+		"Unmarshal must return an UnmarshalTypeError or sonic mismatch error, got: %v", err,
+	)
```

Comment:
🚧  I think we can be cleaner than this
```suggestion
	if json.Implementation == "bytedance/sonic" {
		require.ErrorContains(t, err, "Mismatch type string with value badger", "Unmarshal should return the correct error text for sonic")
	} else {
		require.ErrorAs(t, err, &jsonError, "Unmarshal should return the correct error type for go std encoding/json")
	}
```
Tested with and without sonic, and made sure it's not a false positive.